doc/api/mime.md

```diff
@@ -0,0 +1,231 @@
+# MIME
+
```

Comment:
The docs for this needs to live in the util doc file.

Reply:
still rebasing, see last commit being `wip` and recent comments on the thread